### ISSUE TITLE
Issue with Updating an Attribute form.

### ISSUE THREAD

**Issue by odiea** [opensourcepos/opensourcepos#2201]
Current Master
Mysql 5.7.21
Php 7.1.16
Wamp Server.
I created an Attribute called Colors. I added 4 colors all went well. I tried updating that attribute with more colors and get the following error.

A PHP Error was encountered
Severity: Notice
Message: Undefined index:
Filename: controllers/Attributes.php
Line Number: 74
Backtrace:
File: C:\wamp64\www\osposcurrent\application\controllers\Attributes.php
Line: 74
Function: _error_handler
File: C:\wamp64\www\osposcurrent\public\index.php
Line: 315
Function: require_once
{"success":true,"message":"You have successfully updated attribute 8","id":"8"}

**Comment by jekkos** [comment]
It's on my list. I already quickly tried to reproduce but couldn't find the issue. WIll try again this weekend.

**Comment by odiea** [comment]
So far no change on my side. I have only been testing in incognito or private mode.
This is adding a new attribute.
![2018-11-03 16_50_18-window](https://user-images.githubusercontent.com/10962177/47958095-9d74ef00-df89-11e8-9351-170f8d846447.png)

This is updating a form with a new value. The value does update in the database.
![2018-11-03 16_51_33-](https://user-images.githubusercontent.com/10962177/47958104-bf6e7180-df89-11e8-8ced-f565a74e0d63.png)

**Comment by odiea** [comment]
I downloaded the latest unstable build and tested again.
Updating a Group or a Text or a Date does not give this issue. It appears to be with drop down only at this time.

**Comment by odiea** [comment]
Still no change for the Demo. Has older version.

**Comment by SteveIreland** [comment]
The demo now has the Attributes fix.  It also has the india-gst branch.

**Comment by odiea** [comment]
Yeah and still has the SOS.

**Comment by SteveIreland** [comment]
Sorry.  I must be missing something. Can you explain What constitutes SOS?

**Comment by odiea** [comment]
With Attributes I am still getting the Same Old Stuff happening when updating the Form.

**Comment by odiea** [comment]
The other thing that happens is with the read only field set on the form the attribute type will change after an update is attempted.

**Comment by objecttothis** [comment]
I thought I'd write this here, although it may be worthy of a different issue thread.  I just got around to migrating to the latest bintray and other than a few errors during the migration to 3.3.0 (not attributes migration) everything went as expected. 

Here is the problem.  Now that I have my new handy-dandy attributes I would like to set the type of each attribute to make use of them.  The migration script defaulted everything to the safe choice (TEXT).  I noticed that the attribute_definitions table is such that it's an easy change to change the type from one to the next, however I'm guessing that the act of converting any wild permutations of data formatting would be craziness.  So is it up to the individual to write migration scripts to do the data migration if they want to change from one field-type to the next after either the migration or the creation of the attribute or is there something in the works?
